pysmt/test/smtlib/test_parser_qf_arrays.py

```diff
@@ -16,10 +16,16 @@
 #   limitations under the License.
 #
 import os
+import pytest
+
 from pysmt.test.smtlib.parser_utils import execute_script_fname, SMTLIB_TEST_FILES, SMTLIB_DIR
 
-def test_generator():
+def _get_tests():
     for (logic, f, expected_result) in SMTLIB_TEST_FILES:
         smtfile = os.path.join(SMTLIB_DIR, f)
         if logic.theory.arrays:
-            yield execute_script_fname, smtfile, logic, expected_result
+            yield smtfile, logic, expected_result
```

Comment:
It seems we do not need a generator anymore. What about moving this into `pysmt.test` as :

```python
def smtlib_tests(logic_pred):
    tests = []
    for (logic, f, expected_result) in SMTLIB_TEST_FILES:
        smtfile = os.path.join(SMTLIB_DIR, f)
        if logic_pre(logic):
            tests.append(smtfile, logic, expected_result)
```

Reply:
Done.